core/trino-main/src/test/java/io/trino/block/BlockAssertions.java

```diff
@@ -941,4 +941,13 @@ private static void verifyNullRate(float nullRate)
     {
         verify(nullRate >= 0 && nullRate <= 1, "nullRate %s is not valid", nullRate);
     }
+
+    private static float capNullRate(float nullRate, int positionCount)
```

Comment:
Why is the variable named `positionCount` ? For map and array isn't `offsets[positionCount]` the total length of data in the Block ?

Reply:
> For map and array isn't offsets[positionCount] the total length of data in the Block ?

well yes, if by total length you mean position count of the data/value block

Reply:
Could you call it `entryCount` or `valuesLength` ? At least that's the terminology used in AbstractArrayBlock or AbstractMapBlock to distinguish it from `positionCount`.
Also consider passing down `offsets` and `positionCount` separately to this method if it's always meant to be used in context of Map and Row

Reply:
This method is not specific to `AbstractMapBlock` or `AbstractArrayBlock`. It returns the nullRate that will match exactly the number of null positions for the given position count.